src/decision/justification_heuristic.cpp

```diff
@@ -552,36 +580,81 @@ JustificationHeuristic::findSplitterRec(TNode node, SatValue desiredVal)
   return ret;
 }/* findRecSplit method */
 
-JustificationHeuristic::SearchResult
-JustificationHeuristic::handleAndOrEasy(TNode node, SatValue desiredVal)
+template <bool check2>
+typename JustificationHeuristic<check2>::SearchResult
+JustificationHeuristic<check2>::handleAndOrEasy(TNode node, SatValue desiredVal)
 {
   Assert( (node.getKind() == kind::AND and desiredVal == SAT_VALUE_FALSE) or
           (node.getKind() == kind::OR  and desiredVal == SAT_VALUE_TRUE) );
 
   int numChildren = node.getNumChildren();
   SatValue desiredValInverted = invertValue(desiredVal);
-  for(int i = 0; i < numChildren; ++i) {
-    TNode curNode = getChildByWeight(node, i, desiredVal);
-    if ( tryGetSatValue(curNode) != desiredValInverted ) {
-      SearchResult ret = findSplitterRec(curNode, desiredVal);
-      if(ret != DONT_KNOW) {
-        return ret;
+  if (check2)
+  {
+    for (int i = 0; i < numChildren; i += 2)
+    {
+      std::vector<Node> nodes;
```

Comment:
Instead of creating a vector for every iteration of the outer loop, declare it outside that loop and clear it in here for better performance. I also believe that it should be ok to use `TNode`s here.